libsolidity/interface/SourceReferenceFormatter.cpp

```diff
@@ -55,8 +55,11 @@ void SourceReferenceFormatter::printSourceLocation(SourceLocation const* _locati
 		}
 		if (line.length() > 150)
 		{
-			line = " ... " + line.substr(startColumn, locationLength) + " ... ";
-			startColumn = 5;
+			int len = line.length();
+			line = line.substr(max(0, startColumn - 35), min(startColumn, 35) + min(locationLength + 35, len - startColumn));
+			line = startColumn > 35 ? " ... " + line : line;
```

Comment:
I think I would prefer `if (startColumn > 35) { line = "... " + line; }`, then you could also pull up the conditional adjustment of `startColumn` below.

Also, why do you use `" ... "` here, but `" ..."` below?

Reply:
My first implementation uses both " ... ". However, our current test system will detect any file for tailing whitespaces. So, I have to modify it to avoid the reference file be checked.
I agree. This is ugly.

Reply:
I will use " ... " for both cases and try to avoid the test problem.

Reply:
I would actually argue that you do not need a ` ` after the `...` if nothing else follows.

Reply:
I think if there is nothing follows, there will not print the " ... ", see the test case too_long_line_right_short.sol and its reference.